### PR TITLE
Arduino 7 support

### PR DESCRIPTION
**Pull request by MartinMueller2003**

### CODE REVIEW

File: pio_examples/gui/platformio.ini
Comment (s00500) [inline pio_examples/gui/platformio.ini:46]: should these be in ?

File: pio_examples/gui/platformio.ini
Comment (s00500) [inline pio_examples/gui/platformio.ini:47]: same here

**Review by s00500 — APPROVED** [review]
Hey @MartinMueller2003 

This looks great, seems like some work had to go into document size estimation here...

Lots of print statements but I think I am quite ok with leaving them in the comments

Finally I think the two test flags should be disabled still, otherwise awesome work!

**Comment by MartinMueller2003** [comment]
My personal opinion is that such rests should always be enabled. In this case they allow prospective users to see how the feature works. They don't add a noticeable delay to compiles, dont add a lot of code size and if they are turned off and something breaks, we would never see the issue.

**Comment by s00500** [comment]
Hm ok, I think they do not have to be added into the example as defines though... can as well just be in...

**Comment by MartinMueller2003** [comment]
I very much agree. They were initially implemented as a compile option. IMO, there is no reason whey they cannot always be there and removed the conditional compilation.

**Comment by MartinMueller2003** [comment]
Any chance we can get this merged?

**Comment by s00500** [comment]
sure, should we remove the defines and just have these things in anyway ?

I can make a release too after the merge

**Comment by MartinMueller2003** [comment]
The original intent was to support both ArduinoJson 6.x.x and 7.x.x. It seems to do this now. I would not want to turn the tests off by default. A user can turn off the tests but I am under the impression that the tests are also your "Sales Tool" showing what the library can do. Turning off those tests may result in people not seeing that the functionality is available to them.

**Comment by s00500** [comment]
Turning them off was not really the intent, just removing the defines so they are always included. After all the gui sketch is just an example... what do you think?

Greetings,

Lukas Bachschwell

https://lbs.sh

**Comment by s00500** [comment]
Meanwhile: thanks for the aweslme work here, did not mean to under appreciate that 😅

**Comment by s00500** [comment]
Cool, thanks, Lgtm, will do a release when near a computer again 😅

Thanks so much!

**Comment by MartinMueller2003** [comment]
I just pushed a version without the conditional flags. It now builds all of the new tests.

FYI: 100% of the Arduino IDE examples fail to compile the ESPAsyncWebServer Library. I did NOT change anything there. I am also not a Arduino IDE guy so I do not really know how to fix it.

**Pull request merged**
